### PR TITLE
Derive Generic for AdjacencyMap

### PR DESCRIPTION
**Pull request by srid**

### CODE REVIEW

File: src/Algebra/Graph/NonEmpty/AdjacencyMap/Internal.hs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE DeriveGeneric #-}
```
Comment (snowleopard) [inline src/Algebra/Graph/NonEmpty/AdjacencyMap/Internal.hs]: I'm not sure, but it seems to me that here we derive `Generic` using `GeneralisedNewtypeDeriving`, which is enabled by default. Could you please check?

**Comment by snowleopard** [comment]
@srid Thank you! Shall we do this for non-empty adjacency maps too?

File: src/Algebra/Graph/NonEmpty/AdjacencyMap/Internal.hs (hunk 1 above)
Comment (srid) [reply src/Algebra/Graph/NonEmpty/AdjacencyMap/Internal.hs]: FWIW here's what I get without `DeriveGeneric`:

```
src/Algebra/Graph/AdjacencyMap/Internal.hs:136:51-57: error:
    • Can't make a derived instance of ‘Generic (AdjacencyMap a)’
        (even with cunning GeneralizedNewtypeDeriving):
        You need DeriveGeneric to derive an instance for this class
    • In the newtype declaration for ‘AdjacencyMap’
    |
136 |     adjacencyMap :: Map a (Set a) } deriving (Eq, Generic)
    |                                                   ^^^^^^^

```

**Comment by srid** [comment]
@snowleopard I've added Generic instances for the other two adjacency maps.

File: src/Algebra/Graph/NonEmpty/AdjacencyMap/Internal.hs (hunk 1 above)
Comment (snowleopard) [reply src/Algebra/Graph/NonEmpty/AdjacencyMap/Internal.hs]: Aha, I see, thanks for checking!

**Comment by snowleopard** [comment]
@srid Great, thank you. Merged!

@nobrakal It looks like the performance suite is broken again. Could you please have a look when you have time?

**Comment by nobrakal** [comment]
@snowleopard  Indeed the new `0.4` version prevented the build. This is corrected :)

**Comment by snowleopard** [comment]
Thanks @nobrakal!

**Pull request merged**